holmes/plugins/toolsets/datadog.py

```diff
@@ -133,18 +136,21 @@ def __init__(self):
             tags=[ToolsetTag.CORE],
         )
 
-    def prerequisites_callable(self, config: dict[str, Any]) -> Tuple[bool, str]:
-        if not config:
+    def prerequisites_callable(self) -> Tuple[bool, str]:
+        if not self.config:
             return False, ""
 
         try:
-            dd_config = DatadogConfig(**config)
-            self.dd_api_key = dd_config.dd_api_key
-            self.dd_app_key = dd_config.dd_app_key
+            self.init_config()
```

Comment:
Is it not better to call `toolset.init_config()` within `toolset.check_prerequisites()`?

What happens where there are multiple prerequisites (possible by design)? There is an implicit guarantee thyat prerequisites are run in order but it's just not ideal to have to call `init_config` within a callable prerequisite.

Reply:
Once `init_config` is called inside `check_prerequisites` it is important to catch any config validation error and return an appropriate error message that helps users remediating the error.